src/engy/app_builder.py

```diff
@@ -76,110 +77,190 @@
     produce_files(responses[0])
 
 
 def generate_frontend():
-    frontend_design_prompts = assert_file_exists_and_read(
-        'frontend_design.txt')
-    print('Generate index.html', flush=True)
-    query = f'''Generate "index.html" that connect to the server. The "index.html" design:
+    frontend_design_prompts = assert_file_exists_and_read("frontend_design.txt")
+    print("Generate index.html", flush=True)
+    query = f"""Generate "index.html" that connect to the server. The "index.html" design:
 ```
 {frontend_design_prompts}
 ```
-'''
+"""
     responses, histories = query_llm(
-        query, system_message=HTML_SYSTEM_PROMPT, previous_history=load_history(), filename='index.html')
+        query,
+        system_message=HTML_SYSTEM_PROMPT,
+        previous_history=load_history(),
+        filename="index.html",
+    )
     save_history(histories[0])
     produce_files(responses[0])
 
 
 def stylize_frontend():
-    print('Stylize index.html', flush=True)
-    query = f'''Stylize and make "index.html" beatuful, look production-ready, by using pure CSS website framework "Tailwind CSS".'''
+    print("Stylize index.html", flush=True)
+    query = f"""Stylize and make "index.html" beautiful, look production-ready, by using pure CSS website framework "Tailwind CSS"."""
     responses, histories = query_llm(
-        query, system_message=HTML_SYSTEM_PROMPT, previous_history=load_history(), filename='index.html')
+        query,
+        system_message=HTML_SYSTEM_PROMPT,
+        previous_history=load_history(),
+        filename="index.html",
+    )
     save_history(histories[0])
     produce_files(responses[0])
 
 
 def revise_backend():
-    print('Modify server.py', flush=True)
+    print("Modify server.py", flush=True)
     query = 'Modify "server.py" to also serve "index.html" like a static web server.'
     responses, histories = query_llm(
-        query, system_message=SERVER_SYSTEM_PROMPT, previous_history=load_history(), filename='server.py')
+        query,
+        system_message=SERVER_SYSTEM_PROMPT,
+        previous_history=load_history(),
+        filename="server.py",
+    )
     save_history(histories[0])
     produce_files(responses[0])
 
 
 def generate_run_bash():
-    print('Generate run.sh', flush=True)
+    print("Generate run.sh", flush=True)
     query = f'Generate "run.sh" to pip install required libraries and start the server.'
     responses, histories = query_llm(
-        query, system_message=BASH_SYSTEM_PROMPT, previous_history=load_history(), filename='run.sh')
+        query,
+        system_message=BASH_SYSTEM_PROMPT,
+        previous_history=load_history(),
+        filename="run.sh",
+    )
     save_history(histories[0])
     produce_files(responses[0])
 
 
+def generate_dockerfile():
+    print("Generate Dockerfile", flush=True)
+    current_dir = os.getcwd()
+    current_folder_name = os.path.basename(current_dir)
+    DOCKERFILE = f"""
+<DOCKERFILE>
+FROM python:3.10
```

Comment:
better based on `FROM continuumio/miniconda3` ? easier to install numpy etc. if mentioned

Reply:
sure

Reply:
why is this image better? it doesn't have C++11 compiler to install of dependencies